src/dotnet/Core.Server/Mesh/MeshWatcher.cs

```diff
@@ -96,16 +96,22 @@ protected override async Task OnRun(CancellationToken cancellationToken)
 
                 try {
                     consumeTask ??= changes.Reader.WaitToReadAndConsumeAsync(CancellationToken.None);
-                    var canRead = await consumeTask
-                        .WaitAsync(NodeLocks.UnconditionalCheckPeriod, cancellationToken)
+                    var canReadResult = await consumeTask
+                        .WaitResultAsync(NodeLocks.UnconditionalCheckPeriod, cancellationToken)
                         .ConfigureAwait(false);
-                    // It's important to throw on cancellation here: canRead may return false exactly due to this
-                    cancellationToken.ThrowIfCancellationRequested();
-                    if (!canRead)
-                        throw new OperationCanceledException("Subscription to changes is lost.");
-                    consumeTask = null;
+                    if (canReadResult.HasError) {
+                        if (canReadResult.Error is not TimeoutException)
+                            canReadResult.ThrowIfError();
+                    }
+                    else {
+                        var canRead = canReadResult.Value;
```

Comment:
It's better to use `result.IsValue(value, error)` pattern here.

Reply:
Done. But previous version has already sneaked into the dev branch through another PR.